code/datums/quirks/_quirk.dm

```diff
@@ -48,41 +108,56 @@
+	var/list/where_items_spawned
+	/// If true, the backpack automatically opens on post_add(). Usually set to TRUE when an item is equipped inside the player's backpack.
+	var/open_backpack = FALSE
 
-/datum/quirk/proc/transfer_mob(mob/living/to_mob)
-	quirk_holder.roundstart_quirks -= src
-	to_mob.roundstart_quirks += src
-	if(mob_trait)
-		REMOVE_TRAIT(quirk_holder, mob_trait, ROUNDSTART_TRAIT)
-		ADD_TRAIT(to_mob, mob_trait, ROUNDSTART_TRAIT)
-	quirk_holder = to_mob
-	on_transfer()
-
-/datum/quirk/proc/add() //special "on add" effects
-/datum/quirk/proc/on_spawn() //these should only trigger when the character is being created for the first time, i.e. roundstart/latejoin
-/datum/quirk/proc/remove() //special "on remove" effects
-/datum/quirk/proc/on_process() //process() has some special checks, so this is the actual process
-/datum/quirk/proc/post_add() //for text, disclaimers etc. given after you spawn in with the trait
-/datum/quirk/proc/on_transfer() //code called when the trait is transferred to a new mob
-
-/datum/quirk/process(delta_time)
-	if(QDELETED(quirk_holder))
-		quirk_holder = null
-		qdel(src)
-		return
-	if(quirk_holder.stat == DEAD)
-		return
-	on_process(delta_time)
+/**
+ * Handles inserting an item in any of the valid slots provided, then allows for post_add notification.
+ *
+ * If no valid slot is available for an item, the item is left at the mob's feet.
+ * Arguments:
+ * * quirk_item - The item to give to the quirk holder. If the item is a path, the item will be spawned in first on the player's turf.
+ * * valid_slots - Assoc list of descriptive location strings to item slots that is fed into [/mob/living/carbon/proc/equip_in_one_of_slots]. list(LOCATION_BACKPACK = ITEM_SLOT_BACKPACK)
+ * * flavour_text - Optional flavour text to append to the where_items_spawned string after the item's location.
+ * * default_location - If the item isn't possible to equip in a valid slot, this is a description of where the item was spawned.
+ * * notify_player - If TRUE, adds strings to where_items_spawned list to be output to the player in [/datum/quirk/item_quirk/post_add()]
+ */
+/datum/quirk/item_quirk/proc/give_item_to_holder(quirk_item, list/valid_slots, flavour_text = null, default_location = "at your feet", notify_player = TRUE)
+	if(ispath(quirk_item))
+		quirk_item = new quirk_item(get_turf(quirk_holder))
+
+	var/mob/living/carbon/human/human_holder = quirk_holder
+
+	var/where = human_holder.equip_in_one_of_slots(quirk_item, valid_slots, qdel_on_fail = FALSE) || default_location
+
+	if(where == LOCATION_BACKPACK)
+		open_backpack = TRUE
+
+	if(notify_player)
+		LAZYADD(where_items_spawned, "<span class='boldnotice'>You have \a [quirk_item] [where]. [flavour_text]</span>")
+
+/datum/quirk/item_quirk/post_add()
+	if(open_backpack)
+		var/mob/living/carbon/human/human_holder = quirk_holder
+		SEND_SIGNAL(human_holder.back, COMSIG_TRY_STORAGE_SHOW, human_holder)
```

Comment:
this has big potential to go runtime with human_holder.back is null. In general, this is a pretty meh implementation of signal usage but I think that's a problem with containers in general.

Either way, do check if human_holder.back is null or not, otherwise you get the null.comp_lookup error.

Reply:
Ordinarily human_holder.back will not be null when open_backpack is TRUE.

However, post_add() can be called later on if the mob is missing a client when the quirk was added, and in that timeframe they could have lost their backpack from various means. I'll update it.